src/main/java/com/bakdata/kafka/util/TopicClient.java

```diff
@@ -111,6 +112,7 @@ public void deleteTopic(final String topicName) {
         } catch (final ExecutionException | TimeoutException ex) {
             throw new KafkaAdminException("Failed to delete topic " + topicName, ex);
         }
+        Preconditions.checkState(!this.exists(topicName), "Deletion of topic %s failed", topicName);
```

Comment:
This isn't a precondition. I'd either use Google's `Verify` or a plain if statement + exception.

Reply:
Could you elaborate on the difference? Internally it just checks the condition and then throws

Reply:
see [JavaDoc for Precondition](https://guava.dev/releases/snapshot-jre/api/docs/com/google/common/base/Preconditions.html), under "Non-preconditions":

> It is of course possible to use the methods of this class to check for invalid conditions which are not the caller's fault. Doing so is **not recommended** because it is misleading to future readers of the code and of stack traces. See Conditional failures explained in the Guava User Guide for more advice. Notably, [Verify](https://guava.dev/releases/snapshot-jre/api/docs/com/google/common/base/Verify.html) offers assertions similar to those in this class for non-precondition checks.